### PR TITLE
Add config unit tests

### PR DESCRIPTION
**Pull request by samanthajayasinghe**
### What type of PR is this?

unit-test
### What this PR does / Why we need it?
Adding unit tests to backplane config 

### Special notes for your reviewer
This is an improvement of https://issues.redhat.com/browse/OSD-19188

### Pre-checks (if applicable)

- [x] Ran unit tests locally
- [x] Validated the changes in a cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

File: pkg/cli/config/config_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+		err = config.CheckAPIConnection()
+		if err != nil {
+			t.Failed()
+		}
+
+	})
+
+	t.Run("should fail for empty proxy url", func(t *testing.T) {
+		config := BackplaneConfiguration{URL: "https://api-backplane.apps.openshiftapps.com", ProxyURL: nil}
```
Comment (supreeth7) [inline pkg/cli/config/config_test.go:54]: Must've gotten in there by mistake, but as you already know we shouldn't be even hinting to the API URL which remotely resembles the actual API.

```suggestion
		config := BackplaneConfiguration{URL: "https://dummy-url", ProxyURL: nil}
```

File: pkg/cli/config/config_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+func TestGetBackplaneConnection(t *testing.T) {
+	t.Run("should fail if backplane API return connection errors", func(t *testing.T) {
+
+		svr := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			_, _ = w.Write([]byte("dummy data"))
+		}))
+
+		proxyURL := "http://squid.myproxy.com"
```
Comment (supreeth7) [inline pkg/cli/config/config_test.go:38]:
```suggestion
		proxyURL := "http://dummy-proxy"
```

**Review by supreeth7 — CHANGES_REQUESTED** [review]
Thanks for enhancing our test coverage.
Can you please update the PR description to be more descriptive for posterity?

**Comment by samanthajayasinghe** [comment]
Thanks, @supreeth7, for the review .. sorry for copy/pasting old tests and fixed the issue

**Review by supreeth7 — APPROVED** [review]
/lgtm

**Pull request merged**